packages/teams-js/test/public/video.spec.ts

```diff
@@ -23,38 +23,426 @@ describe('video', () => {
+
+      const messageForRegister = framedPlatformMock.findMessageByFunc('registerHandler');
+      expect(messageForRegister).not.toBeNull();
+      expect(messageForRegister.args.length).toBe(1);
+      expect(messageForRegister.args[0]).toBe('video.newVideoFrame');
+    });
+
+    it('FRAMELESS - should successful register video frame handler', async () => {
+      await framelessPlatformMock.initializeWithContext('sidePanel');
+
+      video.registerForVideoFrame(emptyVideoEffectCallback, videoFrameConfig);
+
+      const messageForRegister = framelessPlatformMock.findMessageByFunc('registerHandler');
+      expect(messageForRegister).not.toBeNull();
+      expect(messageForRegister.args.length).toBe(1);
+      expect(messageForRegister.args[0]).toBe('video.newVideoFrame');
+    });
+
+    it('FRAMED - should successfully invoke video frame event handler', async () => {
+      await framedPlatformMock.initializeWithContext(FrameContexts.sidePanel);
+      let returnedVideoFrame: video.VideoFrame;
+      let handlerInvoked = false;
+
+      let videoEffectCallback = (
+        _frame: video.VideoFrame,
+        _notifyVideoFrameProcessed: () => void,
+        _notifyError: (errorMessage: string) => void,
+      ): void => {
+        handlerInvoked = true;
+        returnedVideoFrame = _frame;
+      };
+
+      video.registerForVideoFrame(videoEffectCallback, videoFrameConfig);
+      const videoFrameMock = {
+        width: 30,
+        height: 40,
+        data: 101,
+      };
+      framedPlatformMock.sendMessage('video.newVideoFrame', videoFrameMock);
+      expect(returnedVideoFrame).toEqual(videoFrameMock);
+      expect(handlerInvoked).toBeTruthy();
+    });
+
+    it('FRAMELESS - should successfully invoke video frame event handler', async () => {
+      await framelessPlatformMock.initializeWithContext(FrameContexts.sidePanel);
+      let returnedVideoFrame: video.VideoFrame;
+      let handlerInvoked = false;
+      //callback
+      let videoEffectCallback = (
+        _frame: video.VideoFrame,
+        _notifyVideoFrameProcessed: () => void,
+        _notifyError: (errorMessage: string) => void,
+      ): void => {
+        handlerInvoked = true;
+        returnedVideoFrame = _frame;
+      };
+      video.registerForVideoFrame(videoEffectCallback, videoFrameConfig);
+      const videoFrameMock = {
+        width: 30,
+        height: 40,
+        data: 101,
+      };
+      framelessPlatformMock.respondToMessage({
+        data: {
+          func: 'video.newVideoFrame',
+          args: [videoFrameMock],
+        },
+      } as DOMMessageEvent);
+      expect(handlerInvoked).toBeTruthy();
+      expect(returnedVideoFrame).toEqual(videoFrameMock);
+    });
+
+    it('FRAMED - should invoke video frame event handler and successfully send videoFrameProcessed', async () => {
+      await framedPlatformMock.initializeWithContext(FrameContexts.sidePanel);
+      const errorMessage = 'error occurs';
```

Comment:
I think the error message could be more descriptive. (Same goes for the other 'error occurs' error messages in the file.) Maybe replace it with something like 'VideoFrame was unexpectedly null'. Different messages for different tests if necessary.

Reply:
This error is coming back from host side. The purpose here is to show to display the same error that we get. That is the very reason, it is named, 'error occurs' because it could be any error and should be displayed correctly. We can talk about this and the other comment on a call.

Reply:
I changed the error name like we talked

Reply:
Discussed offline. error message is now specified.